### DISCUSSION TITLE
Using quarkus native builds

### DISCUSSION THREAD

**Discussion by deathwaiting** [keycloak/keycloak#12754] [Archive]
One of quarkus main killer features is its support for native builds.
I already used those in one of my projects, and despite the headache of some issues, it was a worthy investment.
My startup time is down from like 6 seconds to just millisecods, image size is smaller, all these are important specially on the cloud, when using serverless options like google cloud run.
I was able to run keycloak on cloud run, which run new instances as needed ,
but keycloak is still lacking in this regard, startup time was too long for a cold start, and i was forced to run keycloak continously to avoid it, which seems to eat at my cloud budget -it is a dev environment and my budget is small-.

So, is there any plans or projects to use quarkus native builds on keycloak? Is there any roadblocks  it?

**Comment by heruan** [comment]
Very interested on this topic too, Keycloak (and its operator) are now bottleneck for both image size and startup time of many of ours bundles. With native images we got sizes reduced from ~400MB to ~50MB and startup times from ~2 to ~0.002 sec. It'd be great to have native images also for Keycloak.

Extensions in that case would need to be bundled when building the image, but I guess people that needs extensions and run a native image could build the image with their extensions.
